pkg/operator/certrotation/signer.go

```diff
@@ -87,13 +89,9 @@ func (c RotatedSigningCASecret) EnsureSigningCertKeyPair(ctx context.Context) (*
 
 	// apply necessary metadata (possibly via delete+recreate) if secret exists
 	// this is done before content update to prevent unexpected rollouts
-	if ensureMetadataUpdate(signingCertKeyPairSecret, c.Owner, c.AdditionalAnnotations) && ensureSecretTLSTypeSet(signingCertKeyPairSecret) {
-		actualSigningCertKeyPairSecret, _, err := applyFn(ctx, c.Client, c.EventRecorder, signingCertKeyPairSecret)
-		if err != nil {
-			return nil, false, err
-		}
-		signingCertKeyPairSecret = actualSigningCertKeyPairSecret
-	}
+	needsMetadataUpdate := ensureMetadataUpdate(signingCertKeyPairSecret, c.Owner, c.AdditionalAnnotations)
+	needsTypeChange := ensureSecretTLSTypeSet(signingCertKeyPairSecret)
+	modified = needsMetadataUpdate || needsTypeChange || modified
```

Comment:
Previously, the code called `applyFn` based on the result of an `&&` condition. Now it uses `||`. Can you confirm if this change is intentional?

The new code looks correct to me, but I just wanted to confirm.

Reply:
Yes. Previously we used `&&` to decide if `Update` needs to be called when either type changed or metadata changed. However, `&&` didn't run `ensureSecretTLSTypeSet` if first part was true, and the type didn't change on first iteration, only the metadata.

By splitting this into separate calls we ensure both `ensureMetadataUpdate` and `ensureSecretTLSTypeSet` are called. Now we can use `||` because either of those will require a separate `Update` call - also, we want `|| modified` there too, as the secret may not exist at all.

Do you think this needs to be a comment to make sure it won't get broken later on?